avm/res/network/private-dns-zone/main.bicep

```
@@ -0,0 +1,259 @@
+metadata name = 'Private DNS Zones'
+metadata description = 'This module deploys a Private DNS zone.'
+metadata owner = 'Azure/module-maintainers'
+
+@description('Required. Private DNS zone name.')
+param name string
+
+@description('Optional. Array of A records.')
+param a array?
+
+@description('Optional. Array of AAAA records.')
+param aaaa array?
+
+@description('Optional. Array of CNAME records.')
+param cname array?
+
+@description('Optional. Array of MX records.')
+param mx array?
+
+@description('Optional. Array of PTR records.')
+param ptr array?
+
+@description('Optional. Array of SOA records.')
+param soa array?
+
+@description('Optional. Array of SRV records.')
+param srv array?
+
+@description('Optional. Array of TXT records.')
+param txt array?
+
+@description('Optional. Array of custom objects describing vNet links of the DNS zone. Each object should contain properties \'vnetResourceId\' and \'registrationEnabled\'. The \'vnetResourceId\' is a resource ID of a vNet to link, \'registrationEnabled\' (bool) enables automatic DNS registration in the zone for the linked vNet.')
+param virtualNetworkLinks array?
+
+@description('Optional. The location of the PrivateDNSZone. Should be global.')
+param location string = 'global'
+
+@description('Optional. Array of role assignment objects that contain the \'roleDefinitionIdOrName\' and \'principalId\' to define RBAC role assignments on this resource. In the roleDefinitionIdOrName attribute, you can provide either the display name of the role definition, or its fully qualified ID in the following format: \'/providers/Microsoft.Authorization/roleDefinitions/c2f4ef07-c644-48eb-af81-4b1b4947fb11\'.')
+param roleAssignments roleAssignmentType
+
+@description('Optional. Tags of the resource.')
+param tags object = {}
```

```suggestion
param tags object?
```

In alignment with your DNS resolver PR. Would however require being added as a PSRule workaround if I'm not mistaken, and then require a refresh of the template & readme.

If you feel it's not worth it, feel free to just resolve